src/sagas/bucket.js

```diff
@@ -133,35 +136,41 @@ export function* deleteCollection(getState, action) {
   }
 }
 
-export function* listBucketCollections(getState, action) {
+export function* listBucketCollections(getState: GetStateFn, action: Object): SagaGen {
   const {bid} = action;
   try {
     const bucket = getBucket(bid);
-    const {data} = yield call([bucket, bucket.listCollections]);
-    yield put(listBucketCollectionsSuccess(data));
+    const result = yield call([bucket, bucket.listCollections]);
+    if (result == null) {
+      throw new Error("Empty result.");
+    }
+    yield put(listBucketCollectionsSuccess(result.data));
   } catch(error) {
     yield put(notifyError("Couldn't list bucket collections.", error));
   }
 }
 
-export function* listBucketHistory(getState, action) {
+export function* listBucketHistory(getState: GetStateFn, action: Object): SagaGen {
   const {bid, filters: {resource_name, since}} = action;
   try {
     const bucket = getBucket(bid);
-    const {data} = yield call([bucket, bucket.listHistory], {
+    const result = yield call([bucket, bucket.listHistory], {
       since,
       filters: {
         resource_name,
         exclude_resource_name: "record"
       }
     });
-    yield put(listBucketHistorySuccess(data));
+    if (result == null) {
```

Comment:
So here's why 

![](http://i.imgur.com/8puva0c.png)

I'm afraid we can't do much about it as that's the "expected" behavior in js...